axelrod/strategies/grudger.py

```diff
@@ -293,16 +294,18 @@ def strategy(self, opponent: Player) -> Action:
         The punishment is in the form of 'd' defections followed by a penance of
         'c' consecutive cooperations.
         """
```

Comment:
and on reflection, those cases would never be called.  i will remove the changes.